### DISCUSSION TITLE
auto-deployment from git CLI has stopped working

### DISCUSSION THREAD

**Discussion by notdiogenes** [vercel/community#7026] [Help]
### Summary

Auto-deployment to Vercel from git CLI has stopped working over the past few days. Commits from the the github website still work, and deployments using vercel --prod still work, but git CLI does not.

I thought this might've been because I hit my Hobby tier limit for max number of deployments in one day, so I upgraded to Pro thinking it would fix the issue, but it's been a few days and the problem persists.

Is this a Vercel problem or a github problem?

**Comment by notdiogenes** [comment]
I've given it a couple of weeks to resolve itself, and I'm afraid the problem persists. I'd love to get pointed in the right direction. Thanks!

**Comment by amyegan** [comment]
Hey @notdiogenes. I want to be sure I understand what you mean by "git CLI". Can you walk me through the steps you use?

**Reply by amyegan** [reply]
If you moved a project to your new Pro team, you may need to disconnect and reconnect the repo to update the auth

**Reply by notdiogenes** [reply]
@amyegan Thanks for getting back to me on this. By "git CLI" I'm referring to git add/commit/push, the usual sequence for pushing a change to my repo. Previously doing so would trigger a redeploy on Vercel, but now it's only redeploying for commits through Github's website and via vercel's CLI (vercel --prod).

I don't think I've moved my project to my Pro team — it's still appearing under "Michael's projects" rather than "My Team." I was also experiencing the same behavior while I was still on Hobby, so there's that. :)

![image](https://github.com/vercel/community/assets/168983811/34dd2728-c4d9-4241-8204-90b511e64a05)

**Reply by amyegan** [reply]
Thanks for walking me through what you tried. If you use [Vercel CLI](https://vercel.com/docs/cli) to run [`vercel build`](https://vercel.com/docs/cli/build) on the repo, does that show any errors?

**Reply by notdiogenes** [reply]
Nope, no errors. I hadn't run vercel build on this project, so I vercel pull'd the project settings and allowed it to build, and it was an otherwise clean build.

I should note that my workflow prior to this issue emerging was to simply commit/push to redeploy. I hadn't needed to use vercel --prod. I would prefer not to use vercel --prod for consistency's sake — it means I have to keep my GH repo synced separately from my vercel deploy.

**Reply by amyegan** [reply]
You shouldn't need to use the Vercel CLI for deployments. I just wanted to be sure that it wasn't a configuration error causing the deployment to cancel before attempting to build your site. If you're not seeing any errors then we've ruled that out as the cause.

I have a few more questions to help us narrow down possible causes.
- Does [Vercel for GitHub](https://vercel.com/docs/deployments/git/vercel-for-github) automatic deployment work on other projects or teams you own? 
- Can you confirm that changes you `git add/commit/push` appear in the expected remote repo? (Automatic deployments are based on git commits in the remote repo, so using the CLI instead of a GUI should not make a difference.)
- Did you disconnect and reconnect the repo to update auth for the project? If you hadn't already tried it, does it make a difference when you do?
- Does a newly created project deploy from GitHub successfully? (You can use a template to test this - [vercel.com/templates](https://vercel.com/templates))

**Reply by notdiogenes** [reply]
I appreciate the suggestions. Here's what I've got for you (my additions in bold).

- Does [Vercel for GitHub](https://vercel.com/docs/deployments/git/vercel-for-github) automatic deployment work on other projects or teams you own? **Are we asking whether the automatic deploy on git push works for other projects? If so, then no, it seems all of my projects are affected.**
- Can you confirm that changes you git add/commit/push appear in the expected remote repo? (Automatic deployments are based on git commits in the remote repo, so using the CLI instead of a GUI should not make a difference.) **Yes, they appear in the repo as they should. This is partly what makes me wonder if it's a GitHub issue instead of a Vercel issue.**
- Did you disconnect and reconnect the repo to update auth for the project? If you hadn't already tried it, does it make a difference when you do? **I done this with no luck. (Also, see below.)**
- Does a newly created project deploy from GitHub successfully? (You can use a template to test this - [vercel.com/templates](https://vercel.com/templates)) **Smart suggestion — I tried it, but unfortunately the problem persists. The newly created project deploys only under the circumstances I described prior, i.e., through commits directly from GH's website and via vercel --prod.**

Quite a mystery.

**Reply by MateenPixel** [reply]
I'm unsure if this is exactly my issue as well, but its the closest thing I could find to it. I am using Vercel to host a web application that integrates with the Spotify API. However, I am encountering an issue where the API intermittently stops working, displaying an error indicating a missing client ID. I have observed that running vercel --prod in the terminal temporarily resolves the issue, restoring the functionality of the API token. Unfortunately, this fix only lasts for a few minutes before the problem recurs.

Given that my website is live and relies on the Spotify API for a key feature, I need this functionality to be consistently operational. What steps can I take to ensure that the Spotify API integration remains stable and continuously functional?

**Reply by notdiogenes** [reply]
I'm bumping this topic so that it isn't culled. The problem persists. :)

[a comment by an automated account is not shown]
